File: backend/app/domain/enums.py
from __future__ import annotations

from enum import Enum


class DocumentType(str, Enum):
    QUOTATION = "quotation"
    INVOICE = "invoice"
    RECEIPT = "receipt"
    BILLING = "billing"
    CREDIT_NOTE = "credit_note"
    DEBIT_NOTE = "debit_note"
    DEPOSIT = "deposit"
    PURCHASE_ORDER = "purchase_order"
    RECEIVE = "receive"
    EXPENSE = "expense"
    WITHHOLDING_TAX = "withholding_tax"


class DocumentStatus(str, Enum):
    ACTIVE = "active"
    APPROVED = "approved"
    BILLED = "billed"
    CANCELLED = "cancelled"
    DRAFT = "draft"
    INACTIVE = "inactive"
    INVOICED = "invoiced"
    OVERDUE = "overdue"
    PAID = "paid"
    PARTIAL = "partial"
    PENDING_BILL = "pending_bill"
    PENDING = "pending"
    REJECTED = "rejected"
    SENT = "sent"
    VOID = "void"


class PaymentMethod(str, Enum):
    BANK_TRANSFER = "Bank transfer"
    CASH = "Cash"
    PETTY_CASH = "Petty Cash"
    CHEQUE = "Cheque"
    CREDIT_CARD = "Credit Card"
    CARD = "Card"
    PROMPTPAY = "PromptPay"


class PaymentStatus(str, Enum):
    UNPAID = "unpaid"
    PARTIAL = "partial"
    PAID = "paid"
    OVERDUE = "overdue"
    REFUNDED = "refunded"
    VOID = "void"


class FinancialAccountType(str, Enum):
    BANK = "bank"
    PETTY_CASH = "petty_cash"
    CHEQUE_PAYABLE = "cheque_payable"
    CREDIT_CARD_PAYABLE = "credit_card_payable"
    PAYMENT_GATEWAY = "payment_gateway"


class StockAdjustmentType(str, Enum):
    SALE = "sale"
    PURCHASE = "purchase"
    RETURN = "return"
    WRITE_OFF = "write_off"
    MANUAL = "manual"
    TRANSFER = "transfer"


class TaxMode(str, Enum):
    EXCLUSIVE = "exclusive"
    INCLUSIVE = "inclusive"
    EXEMPT = "exempt"


class NumberingMode(str, Enum):
    CONTINUOUS = "continuous"
    YEAR_MONTH_RESET = "year_month_reset"
    YEARLY_RESET = "yearly_reset"


class AccountingEventType(str, Enum):
    DOCUMENT_ISSUED = "document_issued"
    PAYMENT_RECORDED = "payment_recorded"
    EXPENSE_APPROVED = "expense_approved"
    RECEIVE_CAPTURED = "receive_captured"
    STATUS_CHANGED = "status_changed"

File: frontend/lib/domain/enums.ts
export const DOCUMENT_TYPES = [
  "quotation",
  "invoice",
  "receipt",
  "billing",
  "credit_note",
  "debit_note",
  "deposit",
  "purchase_order",
  "receive",
  "expense",
  "withholding_tax",
] as const;

export type DocumentType = (typeof DOCUMENT_TYPES)[number];

export const DOCUMENT_STATUSES = [
  "active",
  "approved",
  "billed",
  "cancelled",
  "draft",
  "inactive",
  "invoiced",
  "overdue",
  "paid",
  "partial",
  "pending_bill",
  "pending",
  "rejected",
  "sent",
  "void",
] as const;

export type DocumentStatus = (typeof DOCUMENT_STATUSES)[number];

export const PAYMENT_METHODS = [
  "Bank transfer",
  "Bank Transfer",
  "Cash",
  "Petty Cash",
  "Cheque",
  "Credit Card",
  "Card",
  "PromptPay",
] as const;

export type PaymentMethod = (typeof PAYMENT_METHODS)[number];

export const PAYMENT_STATUSES = [
  "unpaid",
  "partial",
  "paid",
  "overdue",
  "refunded",
  "void",
] as const;

export type PaymentStatus = (typeof PAYMENT_STATUSES)[number];

export const FINANCIAL_ACCOUNT_TYPES = [
  "bank",
  "petty_cash",
  "cheque_payable",
  "credit_card_payable",
  "payment_gateway",
] as const;

export type FinancialAccountType = (typeof FINANCIAL_ACCOUNT_TYPES)[number];

export const STOCK_ADJUSTMENT_TYPES = [
  "sale",
  "purchase",
  "return",
  "write_off",
  "manual",
  "transfer",
] as const;

export type StockAdjustmentType = (typeof STOCK_ADJUSTMENT_TYPES)[number];

export const TAX_MODES = ["exclusive", "inclusive", "exempt"] as const;

export type TaxMode = (typeof TAX_MODES)[number];

export const NUMBERING_MODES = ["continuous", "year_month_reset", "yearly_reset"] as const;

export type NumberingMode = (typeof NUMBERING_MODES)[number];

export const ACCOUNTING_EVENT_TYPES = [
  "document_issued",
  "payment_recorded",
  "expense_approved",
  "receive_captured",
  "status_changed",
] as const;

export type AccountingEventType = (typeof ACCOUNTING_EVENT_TYPES)[number];
